src/native/http_request_response.c

```diff
@@ -363,6 +397,8 @@ enum aws_http_outgoing_body_state s_stream_outgoing_body_fn(
 
     (*env)->DeleteGlobalRef(env, jByteBuffer);
 
+    aws_mutex_unlock(&callback->lock);
```

Comment:
Why not combine this unlock with the one in the exception check (line 387) and move it immediately prior to the exception check on line 384?